test/e2e/cpu_assignment_test.go

```diff
@@ -176,41 +151,115 @@ var _ = ginkgo.Describe("CPU Assignment", ginkgo.Serial, ginkgo.Ordered, ginkgo.
 				unallocatedCPUs := availableCPUs.Difference(sharedAllocPre.CPUAssigned)
 				gomega.Expect(unallocatedCPUs.Size()).To(gomega.BeZero(), "available CPUs not in the shared pool: %v", unallocatedCPUs.String())
 
-				fixture.By("creating a guaranteed pod getting exclusive CPUs from the resource claim")
-				exclPod := makeTesterPodWithExclusiveCPUClaim(fxt.Namespace.Name, dracpuTesterImage, exclCPUClaim)
-				exclPod = pinPodToNode(exclPod, targetNode.Name)
-				exclPod, err = e2epod.CreateSync(ctx, fxt.K8SClientset, exclPod)
-				gomega.Expect(err).ToNot(gomega.HaveOccurred(), "cannot create tester pod: %v", err)
+				numCPUs := availableCPUs.Size()
+				// reserve half the CPUs for the shared pool
+				numAvailableForExclusive := numCPUs / 2
+				numPods := numAvailableForExclusive / cpusPerClaim
+				if numPods > maxExclusivePods {
+					numPods = maxExclusivePods
+				}
```

Comment:
this looks OK in theory but I'm curious to see what actually does on GH action runners, which have (had?) like 4 CPUs per instance. Especially because we reserve 2 CPUs.